Add Duco integration documentation (#44468)
home-assistant/home-assistant.github.io: source/_integrations/duco.markdown

---
title: Duco
description: Instructions on how to integrate Duco ventilation with Home Assistant.
ha_release: 2026.4
ha_category:
  - Fan
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@ronaldvdmeer'
ha_domain: duco
ha_platforms:
  - fan
ha_integration_type: hub
ha_quality_scale: bronze
---

The **Duco** {% term integration %} allows you to monitor and control [Duco](https://www.duco.eu/) demand-controlled ventilation (DCV) systems from Home Assistant. Duco produces ventilation boxes for residential buildings that regulate air quality based on CO₂ and humidity sensors. This integration communicates locally with the Duco box over your home network, requiring no cloud connection.

## Supported devices

> This integration communicates with the **DUCO Connectivity Board** (article 0000-4810) via its local REST API over WiFi or Ethernet.

Reviewer: ```suggestion
This integration communicates with the **DUCO Connectivity Board** (article 0000-4810) via its local REST API over Wi-Fi or Ethernet.
```

- DucoBox Energy Premium

## Prerequisites

- A Duco ventilation box with a DUCO Connectivity Board connected to your local network.
- The IP address or hostname of your Duco Connectivity Board.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:
  description: "The IP address or hostname of your DUCO Connectivity Board on the local network."
{% endconfiguration_basic %}

## Supported functionality

The integration creates one device per ventilation node found in your Duco system. Each node represents a physical ventilation unit (such as the main box or a valve).

### Fan

The fan entity lets you control the ventilation speed of a node. You can set the speed as a percentage (mapped to three discrete levels), use the **Auto** preset, or use the on/off controls.

| Control | Action |
|---------|--------|
| Off | Hands control back to Duco (Auto mode). Duco manages the speed automatically. |
| Low (33%) | Low speed — permanent manual override (CNT1). |
| Medium (66%) | Medium speed — permanent manual override (CNT2). |
| High (100%) | High speed — permanent manual override (CNT3). |
| Auto preset | Same as Off: hands control back to Duco. |

> The fan card shows as **off** (grey) when Duco is in automatic control mode, and as **on** when a manual speed is active.

Reviewer: ```suggestion
The fan card shows as **off** (gray) when Duco is in automatic control mode, and as **on** when a manual speed is active.
```

---
title: Duco
description: Instructions on how to integrate Duco ventilation with Home Assistant.
ha_release: 2026.4
ha_category:
  - Fan
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@ronaldvdmeer'
ha_domain: duco
ha_platforms:
  - fan
ha_integration_type: hub
ha_quality_scale: bronze
---

The **Duco** {% term integration %} allows you to monitor and control [Duco](https://www.duco.eu/) demand-controlled ventilation (DCV) systems from Home Assistant. Duco produces ventilation boxes for residential buildings that regulate air quality based on CO₂ and humidity sensors. This integration communicates locally with the Duco box over your home network, requiring no cloud connection.

## Supported devices

This integration communicates with the **DUCO Connectivity Board** (article 0000-4810) via its local REST API over WiFi or Ethernet.

> | Hardware | Status |

Reviewer: We [prefer lists over tables](https://developers.home-assistant.io/docs/documenting/general-style-guide#tables), see [example](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L105).

Author: Good point as well, i'll change this as well. When done i'll commit and push all changes at once.

|----------|--------|
| DUCO Connectivity Board 1.0 | Supported |
| DUCO Connectivity Board 2.0 | Not tested |

Compatible DucoBox models:

- DucoBox Silent Connect
- DucoBox Focus (from firmware version 17xxxx)
- DucoBox Hygro Plus
- DucoBox Energy Comfort / Energy Comfort Plus
- DucoBox Energy Premium

## Prerequisites

- A Duco ventilation box with a DUCO Connectivity Board connected to your local network.
- The IP address or hostname of your Duco Connectivity Board.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:

>   description: "The IP address or hostname of your DUCO Connectivity Board on the local network."

Reviewer: Could you add an example of the IP address or hostname?

Author: IP as example added

{% endconfiguration_basic %}

## Supported functionality

The integration creates one device for the main Duco box. Connected modules (such as CO₂ sensors or humidity sensors) are discovered but not yet exposed as separate devices.

### Fan

The fan entity lets you control the ventilation speed of a node. You can set the speed as a percentage or switch back to automatic mode.

> The fan is always on — turning it off hands control back to Duco (automatic mode), after which the firmware automatically resumes ventilation and the entity switches back to on to reflect the actual state. Because the entity is always on, the turn on action is never triggered.

Reviewer: "Because the entity is always on, the turn on action is never triggered."

Is this correct? Or is it set to medium if turn on is called with no argument? In the code, it says: "Turn on: set a specific speed, or default to medium (CNT2)."

Author: Great catch. Documentation adjusted

> | Action | Result |

Reviewer: We [prefer lists over tables](https://developers.home-assistant.io/docs/documenting/general-style-guide#tables), see [example](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L105).

|--------|--------|
| Turn off | Hands control back to Duco (automatic mode). |
| Speed 33% | Low speed manual override. |
| Speed 66% | Medium speed manual override. |
| Speed 100% | High speed manual override. |
| Auto preset | Same as turn off: hands control back to Duco. |

> When an external device (for example a CO₂ sensor or an RF wall switch) triggers a timed speed override on the Duco box, Home Assistant reflects the current ventilation level as a percentage. These timed states cannot be set from Home Assistant; writing a speed always uses the permanent manual mode.

Reviewer: what do you mean by "uses the permanent manual mode"?

Author: Explained this a bit further

- Return to auto mode when everyone leaves home using a presence-based automation.
- Monitor ventilation activity over time via the logbook.

## Examples

### Activate high ventilation while cooking

This automation switches the ventilation to high speed when the kitchen hood is turned on, and returns it to automatic mode five minutes after the hood is switched off.

```yaml
- alias: "High ventilation while cooking"
  triggers:
    - trigger: state
      entity_id: switch.kitchen_hood
      to: "on"
  actions:
    - action: fan.set_percentage
      target:
        entity_id: fan.living_ventilation
      data:
        percentage: 100

- alias: "Return to auto after cooking"
  triggers:
    - trigger: state
      entity_id: switch.kitchen_hood
      to: "off"
      for: "00:05:00"
  actions:
    - action: fan.turn_off
      target:
        entity_id: fan.living_ventilation
```

### Reduce ventilation when nobody is home

When the last person leaves home, the ventilation hands control back to Duco (automatic mode). When someone returns, it switches to medium speed.

```yaml
- alias: "Ventilation auto mode on leave"
  triggers:

>     - trigger: state

Reviewer: is it on purpose that state is used here and not numeric state? Asking because below, numeric state is used.

Author: Changed it so this should work

---
title: Duco
description: Instructions on how to integrate Duco ventilation with Home Assistant.

> ha_release: 2026.4

Reviewer: ```suggestion
ha_release: 2026.5
```

This integration communicates with the **DUCO Connectivity Board** (article 0000-4810) via its local REST API over Wi-Fi or Ethernet.

Hardware revisions:

- **DUCO Connectivity Board 1.0**: Supported
- **DUCO Connectivity Board 2.0**: Not tested

Compatible DucoBox models:

- DucoBox Silent Connect
- DucoBox Focus (from firmware version 17xxxx)
- DucoBox Hygro Plus
- DucoBox Energy Comfort / Energy Comfort Plus
- DucoBox Energy Premium

## Prerequisites

- A Duco ventilation box with a DUCO Connectivity Board connected to your local network.
- The IP address or hostname of your Duco Connectivity Board.

{% include integrations/config_flow.md %}

{% configuration_basic %}
Host:
  description: "The IP address or hostname of your DUCO Connectivity Board on the local network, for example `192.168.1.10`."
{% endconfiguration_basic %}

## Supported functionality

The integration creates one device for the main Duco box. Connected modules (such as CO₂ sensors or humidity sensors) are discovered but not yet exposed as separate devices.

### Fan

The fan entity lets you control the ventilation speed of a node. You can set the speed as a percentage or switch back to automatic mode.

> The fan is always on — setting the speed to 0% hands control back to Duco (automatic mode), after which the firmware automatically resumes ventilation.

Reviewer: ```suggestion
The fan is always on, setting the speed to 0% hands control back to Duco (automatic mode), after which the firmware automatically resumes ventilation.
```

The following actions are available:

- **Speed 0%**: Hands control back to Duco (automatic mode).
- **Speed 33%**: Low speed manual override.
- **Speed 66%**: Medium speed manual override.
- **Speed 100%**: High speed manual override.

> - **Auto preset**: Same as speed 0%: hands control back to Duco.

Reviewer: ```suggestion
- **Auto preset**: Same as speed 0%; hands control back to Duco.
```

      target:
        entity_id: fan.living_ventilation
      data:
        percentage: 0
```

### Reduce ventilation when nobody is home

When the last person leaves home, the ventilation hands control back to Duco (automatic mode). When someone returns, it switches to medium speed.

```yaml
- alias: "Ventilation auto mode on leave"
  triggers:
    - trigger: numeric_state
      entity_id: zone.home
      below: 1
  actions:
    - action: fan.set_percentage
      target:
        entity_id: fan.living_ventilation
      data:
        percentage: 0

- alias: "Ventilation medium speed on arrive"
  triggers:
    - trigger: numeric_state
      entity_id: zone.home
      above: 0
  actions:
    - action: fan.set_percentage
      target:
        entity_id: fan.living_ventilation
      data:
        percentage: 66
```

## Data updates

The integration {% term polling polls %} the Duco box every 30 seconds.

## Known limitations

- The integration does not yet expose CO₂ and humidity sensor data from connected Duco modules. This is planned for a future update.
- The integration does not support automatic discovery; the IP address or hostname must be entered manually.

> - The Duco box enforces a rate limit of approximately 200 write requests per hour (HTTP 429, error code 18). The integration handles this gracefully, and the firmware resets the quota automatically.

Reviewer: ```suggestion
- The Duco box enforces a rate limit of approximately 200 write requests per day (HTTP 429, error code 18). The integration handles this gracefully, and the firmware resets the quota automatically.
```
In the troubleshooting section, it said the limit is 200 write requests per day.

Author: Sharp, 200 a day was confirmed yesterday by the manufacturer.
